ts_src/transaction_builder.ts

```diff
@@ -264,7 +264,52 @@ export class TransactionBuilder {
         hashType,
       );
     }
+    return signatureHash;
+  }
+
+  attachSignature(
+    vin: number,
+    keyPair: ECPairInterface,
+    signature: Buffer,
+    hashType?: number,
+  ): void {
+    hashType = hashType || Transaction.SIGHASH_ALL;
+    const signatureResolver = (): Buffer => signature;
+    this.__sign(vin, keyPair, signatureResolver, hashType);
+  }
+
+  sign(
+    vin: number,
+    keyPair: ECPairInterface,
+    redeemScript?: Buffer,
+    hashType?: number,
+    witnessValue?: number,
+    witnessScript?: Buffer,
+  ): void {
+    hashType = hashType || Transaction.SIGHASH_ALL;
+    const hashToSign = this.createSignHash(
+      vin,
+      keyPair,
+      redeemScript,
+      hashType,
+      witnessValue,
+      witnessScript,
+    );
+    const signatureResolver = (): Buffer =>
+      keyPair.sign(hashToSign, this.__USE_LOW_R);
+    this.__sign(vin, keyPair, signatureResolver, hashType);
+  }
 
+  private __sign(
+    vin: number,
+    keyPair: ECPairInterface,
+    signatureResolver: () => Buffer,
+    hashType?: number,
```

Comment:
private methods don't need possibly undefined params.

We should always explicitly define these params.